cockatrice/src/deck_loader.cpp

```diff
@@ -108,3 +110,72 @@ DeckLoader::FileFormat DeckLoader::getFormatFromName(const QString &fileName)
     }
     return PlainTextFormat;
 }
+
+bool DeckLoader::saveToStream_Plain(QTextStream &out)
+{
+    // deck header
+    if(!getName().isEmpty())
+        out << "// " << getName() << "\n\n";
+    if(!getComments().isEmpty())
+    {
+        QStringList commentRows = getComments().split(QRegExp("\n|\r\n|\r"));
+        foreach(QString row, commentRows)
+            out << "// " << row << "\n";
+        out << "\n";
+    }
+
+    // loop zones
+    for (int i = 0; i < getRoot()->size(); i++) {
```

Comment:
Does the class support foreach-style iteration? It's nicer to read and less prone to indexing errors than manual looping. If it can be used with stl stuff it might be nice code-wise to break this up so it can be applied in a functional way by mapping over the sequence with `std::transform`